test/gurka/test_landmarks.cc

```diff
@@ -142,3 +148,128 @@ TEST(LandmarkTest, TestParseLandmarks) {
+  // you told it to add (excepting the id because we dont store that).
+
+  for (const auto& e : tile->GetDirectedEdges()) {
+    auto ei = tile->edgeinfo(&e);
+    auto tagged_values = ei.GetTags();
+
+    edge_index = 0;
+    for (const auto& value : tagged_values) {
+      if (value.first != baldr::TaggedValue::kLandmark)
+        continue;
+
+      Landmark landmark(value.second);
+      std::cout << std::setprecision(10) << "landmark: " << landmark.id << " " << landmark.name << " "
+                << static_cast<int>(landmark.type) << " " << landmark.lng << " " << landmark.lat
+                << std::endl;
+
+      // NOTE: in the following checks, all doubles should be rounded to precision 6 to eliminate the
+      // effect of rounding error
+
+      // check data correctness for flexible landmarks
+      std::vector<PointLL> shape = ei.shape();
+      auto point = shape[shape.size() / 2];
+      auto ltype = static_cast<LandmarkType>((edge_index + 1) % invalid_landmark);
+
+      EXPECT_EQ(landmark.id, 0);
+      EXPECT_EQ(landmark.name, std::to_string(edge_index++));
+      EXPECT_EQ(landmark.type, ltype);
+      EXPECT_EQ(round_to(landmark.lng), round_to(point.first));
+      EXPECT_EQ(round_to(landmark.lat), round_to(point.second));
+
+      // check data correctness of the fixed landmark
+      // EXPECT_EQ(landmark.id, 0);
+      // EXPECT_EQ(landmark.name, "fixed landmark");
+      // EXPECT_EQ(landmark.type, LandmarkType::casino);
+      // EXPECT_EQ(round_to(landmark.lng), round_to(lng));
+      // EXPECT_EQ(round_to(landmark.lat), round_to(lat));
+    }
+
+    auto values = ei.GetTaggedValues();
+    edge_index = 0;
+    for (const std::string& v : values) {
+      if (static_cast<baldr::TaggedValue>(v[0]) != baldr::TaggedValue::kLandmark) {
+        continue;
+      }
+
+      Landmark landmark(v.substr(1));
+      std::cout << std::setprecision(10) << "landmark: " << landmark.id << " " << landmark.name << " "
+                << static_cast<int>(landmark.type) << " " << landmark.lng << " " << landmark.lat
+                << std::endl;
+
+      // check data correctness for flexible landmarks
+      std::vector<PointLL> shape = ei.shape();
+      auto point = shape[shape.size() / 2];
+      auto ltype = static_cast<LandmarkType>((edge_index + 1) % invalid_landmark);
+
+      EXPECT_EQ(landmark.id, 0);
+      EXPECT_EQ(landmark.name, std::to_string(edge_index++));
+      EXPECT_EQ(landmark.type, ltype);
+      EXPECT_EQ(round_to(landmark.lng), round_to(point.first));
+      EXPECT_EQ(round_to(landmark.lat), round_to(point.second));
+
+      // check data correctness of the fixed landmark
+      // EXPECT_EQ(landmark.id, 0);
+      // EXPECT_EQ(landmark.name, "fixed landmark");
+      // EXPECT_EQ(landmark.type, LandmarkType::casino);
+      // EXPECT_EQ(round_to(landmark.lng), round_to(lng));
+      // EXPECT_EQ(round_to(landmark.lat), round_to(lat));
+    }
+  }
+}
```

Comment:
newline at end of file missing, I think there's still some unix tools which bark at this..

Reply:
done. I'm suprised pre-commit didn't find this